spyder/widgets/findinfiles.py

```diff
@@ -487,32 +574,30 @@ def get_options(self, all=False):
         else:
             return (path, file_search, exclude, texts, text_re, case_sensitive)
 
-    @Slot()
-    def select_directory(self):
-        """Select directory"""
-        self.parent().redirect_stdio.emit(False)
-        directory = getexistingdirectory(self, _("Select directory"),
-                                         self.path)
-        if directory:
-            directory = to_text_string(osp.abspath(to_text_string(directory)))
-        self.parent().redirect_stdio.emit(True)
-        return directory
+    @property
+    def path(self):
+        return self.path_selection_combo.path
 
     def set_directory(self, directory):
-        self.path = to_text_string(osp.abspath(to_text_string(directory)))
+        self.path_selection_combo.path = to_text_string(osp.abspath(
+                to_text_string(directory)))
+
+    @property
+    def project_path(self):
+        return self.path_selection_combo.project_path
 
     def set_project_path(self, path):
-        self.project_path = to_text_string(osp.abspath(to_text_string(path)))
-        item = self.path_selection_contents.item(PROJECT)
-        item.setFlags(item.flags() | Qt.ItemIsEnabled)
+        self.path_selection_combo.set_project_path(path)
 
     def disable_project_search(self):
-        item = self.path_selection_contents.item(PROJECT)
-        item.setFlags(item.flags() & ~Qt.ItemIsEnabled)
-        self.project_path = None
+        self.path_selection_combo.set_project_path(None)
+
+    @property
+    def file_path(self):
+        return self.path_selection_combo.file_path
 
     def set_file_path(self, path):
-        self.file_path = path
+        self.path_selection_combo.file_path = path

```

Comment:
Should `path` be changed for `osp.abspath(path)` or `to_unicode_from_fs(osp.abspath(path))`?

Reply:
Again, this is only used in one place

https://github.com/jnsebgosselin/spyder/blob/8b9483d72967c4580143fde6adaeec38704f0dbd/spyder/plugins/findinfiles.py#L86

and, since it sets the path of current file in the Editor, I think it's fine as it is, or you can change it to `osp.abspath(path)` if you want to be certain to have a full path.

Reply:
Ok, I'll leave it like that. If it ain't broke, don't fix it.